lib/ws_servers/api/index.js

```diff
@@ -146,27 +168,37 @@ module.exports = class APIWSServer extends WSServer {
     ws.clients = {}
     ws.user = null
 
+    const { dms, affiliateCode } = await this.getUserSettings()
+    ws.algoWorker = new AlgoWorker(
+      { dms, affiliateCode, wsURL: this.wsURL, restURL: this.restURL },
+      this.algos,
+      { ws: send.bind(send, ws) },
+      this.algoDB
+    )
+
     await this.sendInitialConnectionData(ws)
 
     this.exchangeClient.onData((chanID, data) => {
       send(ws, ['data', 'bitfinex', chanID, data])
     })
   }
 
+  async getUserSettings (opts) {
+    const { UserSettings } = this.db
+    const { userSettings } = await UserSettings.getAll()
+
+    const merged = Object.assign(DEFAULT_USER_SETTINGS, userSettings)
```

Comment:
This will mutate the value of `DEFAULT_USER_SETTINGS`, is this expected? We can add a new empty obj as the first arg or use the obj destructuring operator to enforce immutability.

Reply:
good catch!